### PR TITLE
Blend out-of-fold predictions only over members that predicted each row

### LINKED ISSUE #55: `max_num_rows` + calibration: single-split path fits calibration on partially-zeroed OOF probabilities
## Summary

When `max_num_rows` triggers row subsampling, a calibration method is enabled, and the single validation-split path is taken, the out-of-fold probability matrix passed to `_fit_calibration` contains rows that do not sum to 1. For example, with `n_estimators=3`, some rows sum to `0.333` or `0.667`.

The calibrator is therefore fit on a mixture of real predicted probabilities and zero-filled missing predictions, so calibration is silently wrong for exactly the large-dataset case where users are likely to enable `max_num_rows`.

Verified against `main` @ `65aeeed`.

## What's happening

Each ensemble member draws its own row subsample, so each member's validation rows correspond to different absolute row indices.

`predict_oof_proba` writes each member's OOF predictions to the correct places:

```python
outputs_oof[i, val_indices_list[i]] = out_i
```

But on the single-split path it only records member 0's validation indices:

```python
self.oof_val_indices_ = val_indices_list[0]
```

Then `fit()` slices all members using member 0's validation indices:

```python
oof_probs_fit = oof_probs[:, val_idx, :]
```

For members other than 0, most of those rows were never predicted and are still all-zero. Those zeros are included in:

```python
P = np.mean(oof_probs_fit, axis=0)
```

and then passed into `_fit_calibration`, with labels taken from member 0's validation rows.

## Repro

CPU, PyTorch backend, random-init weights. No checkpoint is needed because the issue is index bookkeeping in `fit()`. The spy only captures what `_fit_calibration` receives.

```python
import numpy as np
import torch

from tabfm.src.pytorch import model as pytorch_model
from tabfm.src.classifier_and_regressor import TabFMClassifier

torch.manual_seed(0)
model = pytorch_model.TabFM(
    embed_dim=8,
    max_classes=3,
    col_num_blocks=1,
    col_nhead=2,
    col_num_inds=8,
    row_num_blocks=1,
    row_nhead=2,
    row_num_cls=2,
    icl_num_blocks=1,
    icl_nhead=2,
    ff_factor=2,
    feature_group_size=2,
    is_classifier=True,
)

np.random.seed(0)
X = np.random.rand(40, 3)
y = np.random.randint(0, 2, 40)

captured = {}
orig = TabFMClassifier._fit_calibration

def spy(self, P, y):
    captured["P"] = np.asarray(P).copy()
    return orig(self, P, y)

TabFMClassifier._fit_calibration = spy

clf = TabFMClassifier(
    model=model,
    n_estimators=3,
    batch_size=2,
    random_state=42,
    max_num_rows=30,
    binary_calibration_method="platt",
    num_folds_for_cv=2,
    min_rows_for_single_val_split=5,
)

try:
    clf.fit(X, y)
finally:
    TabFMClassifier._fit_calibration = orig

print(np.round(captured["P"].sum(axis=1), 3))
```

Output:

```text
[0.667 1.    0.667 0.667 0.667 0.667 0.667 0.667 0.333 0.333 0.333 1.
 0.667 0.333 1.   ]
```

12 of 15 rows sum to `k / 3`, where `k` is the number of ensemble members that actually predicted that row.

## Expected behavior

Rows passed to `_fit_calibration` should be valid probability distributions, so each row of `P` should sum to approximately 1.

## Actual behavior

Rows passed to `_fit_calibration` can sum to less than 1 because zero-filled missing OOF predictions are averaged together with real predictions.

## Possible fix direction

The single-split path should either:

1. use a shared validation index set across ensemble members when calibration OOF predictions are needed, or
2. track valid OOF rows per estimator and average only over estimators that actually predicted each row, while aligning `y` to those absolute row indices.

At minimum, calibration should avoid averaging zero-filled missing predictions into `P`.

### PR DESCRIPTION
**Pull request by fus3r**
Fixes #55.

## The bug

When `max_num_rows` is set, each ensemble member subsamples a different set of rows. So in `predict_oof_proba` a member only produces out-of-fold predictions for its own rows, and the rest of its output stays zero.

`fit()` builds the calibration input by averaging over all members, zeros included. That makes rows sum to `k / n_estimators` (k = how many members actually predicted the row) instead of 1, so the calibrator is fit on vectors that aren't valid probabilities. You only hit it with `max_num_rows`, which is the large-data case.

The single-split path made it worse: it kept only member 0's validation indices and sliced every member with them. The multi-fold path averaged over all N rows, including ones a member never sampled.

`enable_nnls` can't be combined with `max_num_rows` (the constructor rejects it), so the NNLS blend never runs into this and I left it as is.

## Repro on main

Random weights, PyTorch, no checkpoint needed:

```python
import numpy as np, torch
from tabfm.src.pytorch import model as pytorch_model
from tabfm.src.classifier_and_regressor import TabFMClassifier

torch.manual_seed(0)
model = pytorch_model.TabFM(
    embed_dim=8, max_classes=3, col_num_blocks=1, col_nhead=2, col_num_inds=8,
    row_num_blocks=1, row_nhead=2, row_num_cls=2, icl_num_blocks=1, icl_nhead=2,
    ff_factor=2, feature_group_size=2, is_classifier=True)

np.random.seed(0)
X, y = np.random.rand(40, 3), np.random.randint(0, 2, 40)

captured = {}
orig = TabFMClassifier._fit_calibration
def spy(self, P, y_):
    captured["P"] = np.asarray(P).copy()
    return orig(self, P, y_)
TabFMClassifier._fit_calibration = spy

clf = TabFMClassifier(
    model=model, n_estimators=3, batch_size=2, random_state=42, max_num_rows=30,
    binary_calibration_method="platt", num_folds_for_cv=2,
    min_rows_for_single_val_split=5)
clf.fit(X, y)
print(np.round(captured["P"].sum(axis=1), 3))
```

On `main` many of those sums come out as 0.333 or 0.667. With the fix they are all 1.0.

## Fix

`predict_oof_proba` now stores which rows each member predicted (`oof_valid_mask_`). `fit()` drops rows that no member predicted, and for the rest it averages each row over just the members that predicted it. Without `max_num_rows` every member predicts every row, the mask is all True, and the result is unchanged.

## Tests

Two tests added: one checks the mask matches each member's predicted rows under subsampling, the other checks the calibration input equals the per-row mean over the predicting members (with a never-predicted row dropped and the labels realigned). `pytest tabfm/src/` is green (74 tests).

### CODE REVIEW

**Comment by fus3r** [comment]
Hi, just following up on this PR. Is there anything you'd like me to add or change before review?

**Comment by erzel** [comment]
Thank you for the contribution and apologies for the delay in reviewing.
We have an internal fix for this which we plan to publish shortly which uses a shared hold-out set instead of CV predictions, instead of fitting calibration on partial ensembles. We believe that would be a better solution since it allows calibration fitting to be closer to inference (where all ensemble members are used not just a subset).

**Comment by fus3r** [comment]
Thanks for the update, and no worries about the delay. The hold-out approach sounds like the cleaner way to handle it. Happy to try the fix when it lands!

**Comment by erzel** [comment]
Sorry for the long delay, but the hold-out set fix is now merged (PR #85). Let us know if you have any more issues.

 Thank you for your patience.
